### ISSUE TITLE
CustomRule to replace a function with a chained method to a Method Call

### ISSUE THREAD

**Issue by ItsReddi** [rectorphp/rector#4225]
# Question
First of all thanks for Rector! 
I am pretty new in rector an a bit lost in the documentation right now.
We are trying to make a custom rule to replace some codeception functions.

Definition would be:
```php
return new RectorDefinition(
            'expect($arg1, $arg2)->N()" will be replaced to "$this->assertN($arg2, $arg1)', [
                new CodeSample(
                    // code before
                    'expect(\'string is not empty\', $object->method())->notEmpty();',
                    // code after
                    '$this->assertNotEmpty($object->method(), \'string is not empty\');'
                ),
            ]
        );
```

Expectations and questions on such a rule:
1. The Node would be a `FuncCall::class`
2. The Name should be in `$node->name`
3. Arguments should be in `$node->args`
4. How to get N()+Arguments?? (N can be Anything e.g. `notEmpty()`, `contains(',')` 
5. Construct `$newMethodName`
5. We would return `new MethodCall($this, $newMethodName, [$arg2,$arg1])`

Real world examples:
```php
expect('string is returned', $user->getEmail())->internalType('string');
expect('string is not empty', $user->getEmail())->notEmpty();
expect('string matches verified.email@testdata.de', $user->getEmail())->equals('verified.email@testdata.de');
expect('string contains the custom seperator', $user->getMailsAsSeperatedString(';'))->contains(';');
``` 
 Would be:
```php
$this->assertIsString($user->getEmail(), 'string is returned');
$this->assertNotEmpty($user->getEmail(), 'string is not empty');
$this->assertEquals('verified.email@testdata.de', $user->getEmail(), 'string matches verified.email@testdata.de');
$this->assertContains(';', $user->getMailsAsSeperatedString(';'), 'string contains the custom seperator');
```

**Comment by TomasVotruba** [comment]
Hi, 

thanks for using Rector to migrate to PHPUnit. 
What actually is the original source... Codeception functions? 

From my experience, the best way to explain rule details is in a code-review of pull-request with a rule. Could you send me pull-request with a rule here? It does not have to be pretty.

Put comments in the code, where exactly you need help. I'll navigate you to finish line :)

<br>

Also, this set might be useful to wider PHP community. Would you like contribute `codeception-to-phpunit` set?

**Comment by ItsReddi** [comment]
Thanks for that suggestion. Its more an generic question than that it is a real task for a rule.
I do not get, how to do 4)

We are not migrating from Codeception to PHPUnit, the mentioned function above seems to be a helper function that was generated somehow im unable to reproduce the real source.
Yes it could be Codeception but Codeception integrates into PHPUnit and integrate Yii2-Codeception in our case.
tl;dr; i do not think that this can be a codeception-to-phpunit set.

But maybe we generate Yii2 upgrade rules (sets) and contribute, if we hit some bc's

**Comment by TomasVotruba** [comment]
I see, no need to contribute right away.

> I do not get, how to do 4)

I'm not sure what you mean. If you send PR with marked line, I'll explain much easily and faster

**Comment by TomasVotruba** [comment]
Closing as follow up PR is missing
